cli/lib/sigstore/cli/id_token.rb

```diff
@@ -85,5 +87,33 @@ def call
         end
       end
     end
+
+    class Buildkite < IdToken
+      def call
+        logger.debug { "looking for OIDC credentials" }
+        unless ENV["BUILDKITE"]
+          logger.debug { "environment doesn't look like Buildkite; giving up" }
+          return
+        end
+
+        raise AmbientCredentialError, "buildkite-agent executable not found" unless buildkite_agent_found?
+
+        request_token
+      end
+
+      private
+
+      def buildkite_agent_found?
+        _, status = Open3.capture2("which buildkite-agent")
+        status.success?
```

Comment:
using `open3` from stdlib is a handy way to avoid using globals and still check the exist status, but I'm open to alternatives if you'd prefer